### PR TITLE
Customer journey: edit password

### PR DESCRIPTION
**Pull request by mrz1880**
This PR is about the page /edit-password
It allows the user to change his password by typing a password and his confirmation.

We externalized the translation

When the user is choosing a password:

- we check that he respects the requirements 
- we check the requirement list accordinhly

When the user is filling the fields: 
we check that the password and the confirmation is identical.

when the user clicks on submit: 

1. we check if the userId is in his token
2. we check if the form is valid
3.  we do a call to update the password 
4. we reconnect the user transparently
5.  we clear the inputs and their validation

As we discussed we create a password repository that will be used on everything that touches the password. We can specify the url within the repository

This branch is coming from alias-112 branch so we'll merge in alias-112 first then merge alias-112 to dev

### CODE REVIEW

**Review by prevostjules — APPROVED** [review]
Overall it seems very very cool . I've left some comments on things we can change now or later but that are not problematic at all ;)

File: src/components/section/UropaHeaderNonConnected.vue [hunk 1]
```diff
@@ -214,10 +215,7 @@ async function submitLogin({
   emailInput: string
   passwordInput: string
 }) {
-  const tokens = await connectApi.login.post({
-    email: emailInput,
-    password: passwordInput,
-  })
+  const tokens = await loginUser(emailInput, passwordInput)
```
[the remark this answers is not in the document]
Comment (mrz1880) [reply src/components/section/UropaHeaderNonConnected.vue:218]: Thanks, i used the method in userAdapter.ts and i guess we should use the method from the adapter in your PR.

File: src/components/basic/InputField.vue [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
-      :value="modelValue"
-      :type="type"
-      :name="name"
-      class="form-field"
-      @input="updateValue"
-    />
-    <slot name="icon" />
-    <slot name="errors" />
-    <slot name="requirements" />
+    <div class="password-input-container">
```
[the remark this answers is not in the document]
Comment (mrz1880) [reply src/components/basic/InputField.vue:4]: No it's not. I'll dig into it before merging. Thanks, good catch

File: src/pages/UropaEditPassword.vue [hunk 3]
```diff
@@ -3,91 +3,232 @@
...
+    specialCharacter: (value) => /[#?!@$%^&*-]/.test(value),
+  },
+  passwordConfirmationInput: {
+    required: helpers.withMessage($t('errors.empty'), required),
+    sameAs: helpers.withMessage(
+      $t('errors.password.sameAs'),
+      sameAs(passwordInput)
+    ),
+  },
+}))
```
[the remark this answers is not in the document]
Comment (mrz1880) [reply src/pages/UropaEditPassword.vue:173]: i agree ! Let's write it down for later

**Pull request merged**